pkg/resource/notebook_instance/sdk.go

```diff
@@ -0,0 +1,576 @@
+		return latest, requeueWaitWhileStopping
+	}
+	input, err := rm.newUpdateRequestPayload(ctx, desired)
+	if err != nil {
+		return nil, err
+	}
+
+	var resp *svcsdk.UpdateNotebookInstanceOutput
+	_ = resp
+	resp, err = rm.sdkapi.UpdateNotebookInstanceWithContext(ctx, input)
+	rm.metrics.RecordAPICall("UPDATE", "UpdateNotebookInstance", err)
+	if err != nil {
+		return nil, err
+	}
+	// Merge in the information we read from the API call above to the copy of
+	// the original Kubernetes object we passed to the function
+	ko := desired.ko.DeepCopy()
+
+	rm.setStatusDefaults(ko)
+	curr := ko.GetAnnotations()
+	if curr == nil {
+		curr = make(map[string]string)
+	}
+	curr["done_updating"] = "true"
+	ko.SetAnnotations(curr)
+	rm.customSetOutputCreateUpdate(ko)
+	return &resource{ko}, nil
+}
+
+// newUpdateRequestPayload returns an SDK-specific struct for the HTTP request
+// payload of the Update API call for the resource
+func (rm *resourceManager) newUpdateRequestPayload(
+	ctx context.Context,
+	r *resource,
+) (*svcsdk.UpdateNotebookInstanceInput, error) {
+	res := &svcsdk.UpdateNotebookInstanceInput{}
+
+	if r.ko.Spec.AcceleratorTypes != nil {
+		f0 := []*string{}
+		for _, f0iter := range r.ko.Spec.AcceleratorTypes {
+			var f0elem string
+			f0elem = *f0iter
+			f0 = append(f0, &f0elem)
+		}
+		res.SetAcceleratorTypes(f0)
+	}
+	if r.ko.Spec.AdditionalCodeRepositories != nil {
+		f1 := []*string{}
+		for _, f1iter := range r.ko.Spec.AdditionalCodeRepositories {
+			var f1elem string
+			f1elem = *f1iter
+			f1 = append(f1, &f1elem)
+		}
+		res.SetAdditionalCodeRepositories(f1)
+	}
+	if r.ko.Spec.DefaultCodeRepository != nil {
+		res.SetDefaultCodeRepository(*r.ko.Spec.DefaultCodeRepository)
+	}
+	if r.ko.Spec.InstanceType != nil {
+		res.SetInstanceType(*r.ko.Spec.InstanceType)
+	}
+	if r.ko.Spec.LifecycleConfigName != nil {
+		res.SetLifecycleConfigName(*r.ko.Spec.LifecycleConfigName)
+	}
+	if r.ko.Spec.NotebookInstanceName != nil {
+		res.SetNotebookInstanceName(*r.ko.Spec.NotebookInstanceName)
+	}
+	if r.ko.Spec.RoleARN != nil {
+		res.SetRoleArn(*r.ko.Spec.RoleARN)
+	}
+	if r.ko.Spec.RootAccess != nil {
+		res.SetRootAccess(*r.ko.Spec.RootAccess)
+	}
+	if r.ko.Spec.VolumeSizeInGB != nil {
+		res.SetVolumeSizeInGB(*r.ko.Spec.VolumeSizeInGB)
+	}
+
+	return res, nil
+}
+
+// sdkDelete deletes the supplied resource in the backend AWS service API
+func (rm *resourceManager) sdkDelete(
+	ctx context.Context,
+	r *resource,
+) (latest *resource, err error) {
+	rlog := ackrtlog.FromContext(ctx)
+	exit := rlog.Trace("rm.sdkDelete")
+	defer exit(err)
+	//This will avoid exponential backoff
+	if err = rm.requeueUntilCanModify(ctx, r); err != nil {
+		return r, err
+	}
+
+	rm.customPreDelete(r)
+	input, err := rm.newDeleteRequestPayload(r)
+	if err != nil {
+		return nil, err
+	}
+	var resp *svcsdk.DeleteNotebookInstanceOutput
+	_ = resp
+	resp, err = rm.sdkapi.DeleteNotebookInstanceWithContext(ctx, input)
```

Comment:
Do we also need something like [this ](https://github.com/aws-controllers-k8s/sagemaker-controller/blob/bb7eb30d6802ba6caf7e644f216e0c645bef527e/pkg/resource/model_package/sdk.go#L978) here?